src/integ_test/java/games/strategy/engine/lobby/server/db/BannedMacControllerTest.java

```diff
@@ -0,0 +1,72 @@
+package games.strategy.engine.lobby.server.db;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.sql.Timestamp;
+import java.time.Instant;
+
+import org.junit.Test;
+
+import games.strategy.util.MD5Crypt;
+import games.strategy.util.ThreadUtil;
+import games.strategy.util.Tuple;
+import games.strategy.util.Util;
+
+public class BannedMacControllerTest {
+
+  private final BannedMacController controller = new BannedMacController();
+
+  @Test
+  public void testBanMacForever() {
+    final String hashedMac = banMac(null);
+    final Tuple<Boolean, Timestamp> macDetails = controller.isMacBanned(hashedMac);
+    assertTrue(macDetails.getFirst());
+    assertNull(macDetails.getSecond());
+  }
+
+  @Test
+  public void testBanMac() {
+    final Instant banUntil = Instant.now().plusSeconds(2L);
+    final String hashedMac = banMac(banUntil);
+    final Tuple<Boolean, Timestamp> macDetails = controller.isMacBanned(hashedMac);
+    assertTrue(macDetails.getFirst());
+    assertEquals(banUntil, macDetails.getSecond().toInstant());
+    while (banUntil.isAfter(Instant.now())) {
+      ThreadUtil.sleep(100);
+    }
```

Comment:
An improvement we can look at in a future PR (or here if you're so inclined :smile:) is to avoid the dependency on wall time in these tests.  Not only do we have several tests that take 2+ seconds to run, but they still have a race condition; if the build environment were to be running dead slow for some reason, they might fail.

It looks like the problem (in this case) is that the `BannedMacController#isMacBanned()` method depends on `Instant#now()`.  We could extract a package-private overload of `isMacBanned()` that accepted an additional `Instant` parameter representing "now" so the tests could provide that value and run at full speed without having to wait for a certain amount of time to pass.  For example:

```java
public class BannedMacController {
  // . . .

  public Tuple<Boolean, Timestamp> isMacBanned(final String mac) {
    return isMacBanned(mac, Instant.now());
  }

  @VisibleForTesting
  Tuple<Boolean, Timestamp> isMacBanned(final String mac, final Instant now) {
    // same as existing implementation except that Instant.now() replaced with parameter "now"
  }
}
```

A similar change could be made for the other controllers that also depend on `Instant#now()`.

Reply:
Agreed that depending on execution time is not a good idea, however I have a better idea, might adress in this PR if it works out as expected.

Reply:
Didn't work out as expected...
Wil adress in a future PR